kernel: remove mac-address-increment support

neheb opened the pull request:
Deprecated and replaced treewide with mac-base. Remove to avoid modern usage.

ping @rmilecki @Ansuel

neheb committed:
kernel: remove mac-address-ascii dts support

Deprecated and replaced treewide with mac-base. Remove to avoid modern
usage.

Signed-off-by: Rosen Penev <rosenp@gmail.com>

Ansuel commented:
Must be a miracle that we finally manage to eradicate this! Thanks for all the conversion work.

DragonBluep commented on target/linux/generic/pending-5.15/683-of_net-add-mac-address-to-of-tree.patch:
```diff
@@ -45,11 +45,11 @@ property. This way, the MAC address can be accessed using procfs.
  
 +	of_add_mac_address(np, addr);
- 	return ret;
+ 	return of_get_mac_addr_nvmem(np, addr);
```
I think the logic is wrong here. This may break the label mac hack. Suggested change: https://github.com/DragonBluep/openwrt/commit/c3a6146b4b4e59bdfb8e39abbb5861698c7da1d0

neheb replied:
Pulled.

DragonBluep replied:
If you like, please keep your signature. Thanks for pushing it forward. It's really a hard job to convert all targets.

neheb replied:
sure.

neheb committed:
kernel: of: remove mac-address-increment support hack

The MAC address increment has been replaced by the new "mac-base"
NVMEM fixed layout. This old implementation can be removed now.

Signed-off-by: Shiji Yang <yangshiji66@qq.com>
Signed-off-by: Rosen Penev <rosenp@gmail.com>

neheb commented:
ping @hauke
